internal/oci/uvm.go

```diff
@@ -434,5 +439,21 @@ func UpdateSpecFromOptions(s specs.Spec, opts *runhcsopts.Options) specs.Spec {
 			s.Annotations[key] = value
 		}
 	}
-	return s
+
+	// expand annotations
+	for key, exps := range annotations.AnnotationExpansions {
```

Comment:
`UpdateSpecFromOptions` is only called once (in `createInternal` in `cmd\containerd-shim-runhcs-v1\service_internal.go`), so this seemed like an appropriate place to add this logic, but the fact that it lived in `oci/uvm` and deals mostly with UVM concerns makes want to move it elsewhere. 
Thoughts?

Reply:
I don't think that this makes sense to put in this function. The point of this function is to overwrite the spec annotations based on shim options in the toml. This could go in your `specToUVMCreateOptionsCommon` function and you could also convert the `handleAnnotationPreferredRootFSType` `handleAnnotation......` functions to be expanded annotations instead.

Reply:
The `handleAnnotation*` functions have some logic associated with them to go from one annotation to setting several different `*uvm.Option*COW` setting, so I think using expanded annotations would convolute the code flow. 
I will put a new `processAnnotations` function to handle this, since if I am moving it out, I want it someplace that isnt associated with uVMs at all (gMSAs apply to process isolated containers too, I think ...)

Reply:
I see what you're saying and maybe this is future work to look at, but I mean each of those `*uvm.Options*COW` settings map to an annotation, so essentially the handle annotation functions do something very similar to what the annotation expansions do.

Reply:
Oh, I misunderstood. Yeah, we could have a similar expansion mechanism from annotations to `uvm.Options*COW` settings.
Or since the handle functions all look the same, make a mapping from annotations to functions that operate on an `*uvm.Options*COW` 
I don't think I could get that in this PR tho 😅